### PR TITLE
Implemented the background image component to support stretch and repeat mode

### PR DESCRIPTION
**Pull request by IbrahimSulai**
Worked to resolve the issue: https://github.com/Imaginea/AdaptiveCards/issues/36

Steps to test the PR:
1. Launch the App.
2. Navigate to "BackgroungImage.RepeatHorizontally.json" from the visualiser.
3. Expected: The backgroundImage will repeat horizontally.

As same as above, we can test  RepeatVertically, Repeat and Stretch mode for backgroundImage.

### CODE REVIEW

File: source/reactnative/src/utils/backgroung-image.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        let error = { "error": Enums.ValidationError.InvalidPropertyValue, "message": `Not able to load the image` };
+        onParseError(error);
+    }
+
+    /**
+	 * @description
+     * This function will return the vertical alignment of the image
+     * 
+     */
+    getImageVerticalyAlignment = () => {
```
Comment (pragadeeshk) [inline source/reactnative/src/utils/backgroung-image.js]: Nit: change name method name to getImageVerticalAlignment

File: source/reactnative/src/adaptive-cards.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ import React from 'react';
 import {
 	StyleSheet,
 	Text,
+	View,
 	ScrollView,
 	ImageBackground
```
Comment (BalajiR) [inline source/reactnative/src/adaptive-cards.js]: Remove unused imports.

File: source/reactnative/src/adaptive-cards.js [hunk 3]
```diff
@@ -20,6 +21,7 @@ import * as Utils from './utils/util';
 import { SelectAction } from './components/actions';
 import ResourceInformation from './utils/resource-information';
 import { ContainerWrapper } from './components/containers/';
+import { BackgroundImage } from './utils/backgroung-image';
```
Comment (BalajiR) [inline source/reactnative/src/adaptive-cards.js]: Move the BackgroundImage components from utils into components directory (under elements).

File: source/reactnative/src/adaptive-cards.js [hunk 4]
```diff
@@ -105,9 +106,10 @@ export default class AdaptiveCards extends React.Component {
 		if (!Utils.isNullOrEmpty(this.payload.backgroundImage)) {
 			let backgroundImage = Utils.getImageUrl(this.payload.backgroundImage)
```
Comment (BalajiR) [inline source/reactnative/src/adaptive-cards.js]: Remove unused variable.

File: source/reactnative/src/utils/backgroung-image.js [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+import * as Utils from './util';
+import PropTypes from 'prop-types';
+import { InputContextConsumer } from './context';
+
+export class BackgroundImage extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.backgroundImage = props.backgroundImage;
+        this.backgroundImageModeEnumValue = Utils.parseHostConfigEnum(
```
Comment (BalajiR) [inline source/reactnative/src/utils/backgroung-image.js]: Move these host config related parsing into new method (say, parseHostConfig).

File: source/reactnative/src/adaptive-cards.js (hunk 2 above)
Comment (IbrahimSulai) [reply source/reactnative/src/adaptive-cards.js]: Done

File: source/reactnative/src/adaptive-cards.js (hunk 3 above)
Comment (IbrahimSulai) [reply source/reactnative/src/adaptive-cards.js]: Done

File: source/reactnative/src/adaptive-cards.js (hunk 4 above)
Comment (IbrahimSulai) [reply source/reactnative/src/adaptive-cards.js]: Done

File: source/reactnative/src/utils/backgroung-image.js (hunk 5 above)
Comment (IbrahimSulai) [reply source/reactnative/src/utils/backgroung-image.js]: Done

**Pull request merged**